third_party/txt/benchmarks/skparagraph_benchmarks.cc

```diff
@@ -0,0 +1,302 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <sstream>
+#include "flutter/fml/command_line.h"
+#include "flutter/fml/logging.h"
+#include "flutter/third_party/txt/tests/txt_test_utils.h"
+#include "third_party/benchmark/include/benchmark/benchmark_api.h"
+#include "third_party/icu/source/common/unicode/unistr.h"
+#include "third_party/skia/include/core/SkBitmap.h"
+#include "third_party/skia/include/core/SkCanvas.h"
+#include "third_party/skia/include/core/SkColor.h"
+#include "third_party/skia/modules/skparagraph/include/Paragraph.h"
+#include "third_party/skia/modules/skparagraph/include/ParagraphBuilder.h"
+#include "third_party/skia/modules/skparagraph/include/TypefaceFontProvider.h"
+#include "third_party/skia/modules/skparagraph/utils/TestFontCollection.h"
+
+namespace sktxt = skia::textlayout;
+static std::unique_ptr<SkCanvas> canvas = nullptr;
+static std::unique_ptr<SkBitmap> bitmap = nullptr;
+static sk_sp<sktxt::TestFontCollection> fontCollection = nullptr;
+
+static void initialize() {
+  if (fontCollection == nullptr) {
+    fontCollection = sk_make_sp<sktxt::TestFontCollection>(txt::GetFontDir());
+  }
+  if (canvas == nullptr) {
+    bitmap = std::make_unique<SkBitmap>();
+    bitmap->allocN32Pixels(1000, 1000);
+    canvas = std::make_unique<SkCanvas>(*bitmap);
+    canvas->clear(SK_ColorWHITE);
+  }
+}
+
+static void BM_SkParagraphShortLayout(benchmark::State& state) {
+  initialize();
+
+  const char* text = "Hello World";
+  sktxt::ParagraphStyle paragraph_style;
+  sktxt::TextStyle text_style;
+  text_style.setFontFamilies({SkString("Roboto")});
+  text_style.setColor(SK_ColorBLACK);
+  auto builder = sktxt::ParagraphBuilder::make(paragraph_style, fontCollection);
+  builder->pushStyle(text_style);
+  builder->addText(text);
+  builder->pop();
+  auto paragraph = builder->Build();
+  while (state.KeepRunning()) {
+    paragraph->markDirty();
```

Comment:
Here an elsewhere, maybe pause timing while marking it dirty so we only benchmark the layout.